### PR TITLE
Added Breit-Wigner distribution

### PR DESCRIPTION
**Pull request by andrewfowlie**
I've added the relativistic Breit-Wigner distribution to `scipy.stats`. The `_pdf` and `_cdf` are analytic; other attributes are numerical and provided by the inherited `rv_continuous` class.

As for the other distributions, I define a `breit_wigner_gen` class, and then a `breit_wigner` instance of that class, which should be imported by a user.

### CODE REVIEW

**Comment by innisfree** [comment]
A few issues I need to fix:

1. ERROR: objects in scipy.stats.__all__ but not in refguide:: breit_wigner
2.  print 'number of continuous distributions:', len(dist_continu)

Expected:

    number of continuous distributions: 94

Got:

    number of continuous distributions: 95

3. All my doctests are broken (though work on my system)

File: scipy/stats/_continuous_distns.py [hunk 1]
```diff
@@ -5195,6 +5195,156 @@ def _entropy(self, beta):
...
+            Mass of resonance        
+        width : float
+            Width of resonance
+
+        Returns
+        -------
+
+        _argcheck :  bool
+            Whether shape parameters are legitimate
+        """
```
Comment (ev-br) [inline scipy/stats/_continuous_distns.py]: Nitpick: There's no need to add a full-fledged docstring to a private, computational method. A simple comment would suffice IMO (and even then, the one-line code is self-documenting enough, I'd think)

File: scipy/stats/_continuous_distns.py [hunk 2]
```diff
@@ -5195,6 +5195,156 @@ def _entropy(self, beta):
...
+
+        Returns
+        -------
+
+        pdf :  float
+            PDF of Breit-Wigner distribution
+
+        >>> breit_wigner(mass=125., width=0.05).pdf(125.)
+        12.732396211295313
+        """
```
Comment (ev-br) [inline scipy/stats/_continuous_distns.py]: Ditto, these docstrings will not be visible unless you look at the source code. Description of the physical parameters, `mass`, `width` and explanations should go to the class docstsring.

File: scipy/stats/_continuous_distns.py [hunk 3]
```diff
@@ -5195,6 +5195,156 @@ def _entropy(self, beta):
...
+        arg_1 = complex(-1)**(1. / 4.) / (-1j + alpha)**0.5 * m / mass
+        arg_2 = complex(-1)**(3. / 4.) / (1j + alpha)**0.5 * m / mass
+
+        shape = -1j * np.arctan(arg_1) / (-1j + alpha)**0.5 - np.arctan(arg_2) / (1j + alpha)**0.5
+        norm = complex(-1)**(1. / 4.) * k / (2. * alpha * mass**3)
+
+        cdf_ = shape * norm
+        cdf_ = cdf_.real
+
+        return cdf_
```
Comment (ev-br) [inline scipy/stats/_continuous_distns.py]: This is a direct transcription of the output of Wolfram Alpha, is it? :-).

I'm pretty sure the formulas can be simplified for real-valued mass and width.

**Comment by ev-br** [comment]
First and foremost, my question about the use case (https://github.com/scipy/scipy/issues/6414) still stands.

On a technical level, this is a very good start!
Assuming there is a use case, this needs some work. See, e.g. https://github.com/scipy/scipy/pull/6030 about several places a distribution needs to be registered with (`__init__`, `_distr_param.py` etc).

File: scipy/stats/_continuous_distns.py [hunk 4]
```diff
@@ -5195,6 +5195,156 @@ def _entropy(self, beta):
...
+    
+    >>> BW.interval(0.95)
+    (53.719690871945822, 159.27077425799101)
+    
+    Moments:
+    
+    >>> [BW.moment(n) for n in range(1, 6)]
+    [122.11538219739913, 15674.920254744189, inf, 244138132.80994478, 80583978992.641495]
+    
+    """
```
Comment (ev-br) [inline scipy/stats/_continuous_distns.py]: Have a look at the docstrings of other distributions, they have magic substitution keys (compare to online docs, where the keys are expanded). I probably makes sense to reuse the standard format.

Comment (ev-br) [inline scipy/stats/_continuous_distns.py]: The docstring also needs a See Also section with a reference to `halfcauchy`.

**Comment by frederikfaye** [comment]
Any news on this? Would love this feature!

**Comment by rgommers** [comment]
The unanswered question is about whether `cauchy` is enough. From gh-6414:

_And indeed, as a function of the square of the 4-momentum, E^2, Breit-Wigner formula is almost a Cauchy distribution, the only difference being that the support of the distribution is [0, \inf], so it's a halfcauchy._

_Can you sketch a use case where a distribution of E^2 is not enough and you really need a distribution in terms of E itself?_

Once that's decided, this looks close to good to go.

File: scipy/stats/_continuous_distns.py [hunk 5]
```diff
@@ -5195,6 +5195,130 @@ def _entropy(self, beta):
...
+    >>> BW = breit_wigner(mass=125., width=10.)
+    
+    >>> mass = np.linspace(0., 200., 500)
+    >>> pdf = [BW.pdf(m) for m in mass]
+    >>> pdf_plot = plt.plot(mass, pdf)
+    
+    >>> samples = BW.rvs(1000)
+    >>> hist_plot = plt.hist(samples, normed=True, bins=100, range=[0, 200])
+    
+    >>> plt.show()
```
Comment (ev-br) [inline scipy/stats/_continuous_distns.py]: Explicit example largely duplicates the autogenerated. Suggest deduplicating.

File: scipy/stats/_continuous_distns.py [hunk 6]
```diff
@@ -5195,6 +5195,130 @@ def _entropy(self, beta):
...
+    -----
+    We require mass > 0 and width > 0.
+
+    The width -> 0 limit is well-defined mathematically - it's proportional
+    to a Dirac function. In the m -> 0 limit, the pdf vanishes.
+    
+    The CDF was found by Mathematica:
+
+    pdf = k/((m^2 - mass^2)^2 + mass^4*alpha^2)
+    cdf = Integrate[pdf, m]
```
Comment (ev-br) [inline scipy/stats/_continuous_distns.py]: This comment belongs to the comment of `_cdf`, should not be in the main docstring.

File: scipy/stats/_continuous_distns.py [hunk 7]
```diff
@@ -5195,6 +5195,130 @@ def _entropy(self, beta):
...
+        Width of resonance 
+   
+    %(before_notes)s
+        
+    Notes
+    -----
+    We require mass > 0 and width > 0.
+
+    The width -> 0 limit is well-defined mathematically - it's proportional
+    to a Dirac function. In the m -> 0 limit, the pdf vanishes.
```
Comment (ev-br) [inline scipy/stats/_continuous_distns.py]: Dirac delta function

File: scipy/stats/_continuous_distns.py [hunk 8]
```diff
@@ -5195,6 +5195,130 @@ def _entropy(self, beta):
...
+    
+    The CDF was found by Mathematica:
+
+    pdf = k/((m^2 - mass^2)^2 + mass^4*alpha^2)
```
Comment (ev-br) [inline scipy/stats/_continuous_distns.py]: Most other distribution have TeX formulas here.

File: scipy/stats/_continuous_distns.py [hunk 9]
```diff
@@ -5195,6 +5195,130 @@ def _entropy(self, beta):
...
+        0.0
+        """
+        alpha_ = width / mass
+        gamma_ = mass**2 * (1. + alpha_**2)**0.5
+        k = 2.**(3. / 2.) * mass**2 * alpha_ * gamma_ / (pi * (mass**2 + gamma_)**0.5)
+        
+        arg_1 = complex(-1)**(1. / 4.) / (-1j + alpha_)**0.5 * m / mass
+        arg_2 = complex(-1)**(3. / 4.) / (1j + alpha_)**0.5 * m / mass
+
+        shape_ = -1j * np.arctan(arg_1) / (-1j + alpha_)**0.5 - np.arctan(arg_2) / (1j + alpha_)**0.5
```
Comment (ev-br) [inline scipy/stats/_continuous_distns.py]: I'm pretty sure this formula could be simplified further to avoid complex arithmetics.

**Review by ev-br — CHANGES_REQUESTED** [review]
a few minor comments (assuming that a use case is going to materialize).

**Comment by frederikfaye** [comment]
> The unanswered question is about whether `cauchy` is enough.

I think it is, see https://github.com/scipy/scipy/issues/6414#issuecomment-522715954.

**Comment by andrewfowlie** [comment]
I don't have time for this, sorry, but would happy to see it make it into scipy :)

**Pull request closed without merging**